src/app/(dashboard)/recipes/utils.ts

```diff
@@ -21,8 +21,14 @@ export const createIngredientRowsStructure = (
     ingredients?: RecipeFormIngredient[]
 ) =>
     ingredientIds.map((id) => {
-        if (!ingredients) return { id }
-        return ingredients.find((ingredient) => ingredient.id === id) ?? { id }
+        if (!ingredients) return { id, amount: '', unit: '' }
+        return (
+            ingredients.find((ingredient) => ingredient.id === id) ?? {
+                id,
+                amount: '',
+                unit: '',
+            }
```

Comment:
Tu chyba jednak trzeba zrobić coś takiego, ale niewiem wsm

```suggestion
            ingredients.find((ingredient) => ingredient.id === id) ?? {
                ...ingredient,
                id,
            }
```